diff -Nru ../evdi-1.0.335/evdi_encoder.c ./evdi_encoder.c
--- ../evdi-1.0.335/evdi_encoder.c	2015-11-29 13:40:58.000000000 +0100
+++ ./evdi_encoder.c	2016-04-04 14:28:42.261056299 +0200
@@ -10,6 +10,7 @@
  * more details.
  */
 
+#include <linux/version.h>
 #include <drm/drmP.h>
 #include <drm/drm_crtc.h>
 #include <drm/drm_crtc_helper.h>
@@ -69,7 +70,11 @@
     return NULL;
   }
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,5,0))
   status = drm_encoder_init(dev, encoder, &evdi_enc_funcs, DRM_MODE_ENCODER_TMDS);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,5,0)
+  status = drm_encoder_init(dev, encoder, &evdi_enc_funcs, DRM_MODE_ENCODER_TMDS, dev_name(dev->dev));
+#endif
   EVDI_DEBUG("drm_encoder_init: %d\n", status);
   drm_encoder_helper_add(encoder, &evdi_helper_funcs);
   encoder->possible_crtcs = 1;
diff -Nru ../evdi-1.0.335/evdi_modeset.c ./evdi_modeset.c
--- ../evdi-1.0.335/evdi_modeset.c	2015-11-29 13:40:58.000000000 +0100
+++ ./evdi_modeset.c	2016-04-04 14:26:26.845050291 +0200
@@ -144,9 +144,12 @@
 
   drm_mode_create_dirty_info_property(dev);
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0)
-  drm_dev_set_unique(dev, "%s", dev_name(dev->dev));
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0) && LINUX_VERSION_CODE < KERNEL_VERSION(4,5,0))
+ 	drm_dev_set_unique(dev, "%s", dev_name(dev->dev));
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,5,0)
+	drm_dev_set_unique(dev, dev_name(dev->dev));
 #endif
+
   evdi_crtc_init(dev);
 
   encoder = evdi_encoder_init(dev);
